pkg/endpoint/policy.go

```diff
@@ -448,28 +449,27 @@ func (e *Endpoint) regenerateConsumable(owner Owner, labelsMap *policy.IdentityC
 	}
 
 	// Garbage collect all unused entries
-	for _, val := range c.Consumers {
-		if val.DeletionMark {
-			val.DeletionMark = false
-			c.BanConsumerLocked(val.ID)
+	for val, ok := range c.Consumers {
```

Comment:
this should not be `val, ok` it should be something like `identity, deny` and you can swap the if statement bellow

Reply:
This is addressed by a later commit in the PR.